Kernel/TimerQueue.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <Kernel/Arch/i386/PIT.h>
+
+struct Timer {
+    u64 id;
+    u64 expires;
+    void (*callback)();
+    bool operator<(const Timer& rhs) const
+    {
+        return expires < rhs.expires;
+    }
+    bool operator>( const Timer& rhs) const
+    {
+        return expires > rhs.expires;
+    }
+
+    bool operator==( const X& rhs) const
+    {
+        return id == rhs.id;
+    }
+};
+
+enum TimeUnit {
+    MS = TICKS_PER_SECOND / 1000,
+    S = TICKS_PER_SECOND,
+    M = TICKS_PER_SECOND * 60
+};
+
+
+class TimerQueue {
```

Comment:
This is a pretty roundabout way to basically add a lot of global functions.
Instead of having a TimerQueue class with nothing but public static functions, let's use a singleton object:

```cpp
// In header file:
class TimerQueue {
public:
    static TimerQueue& the();
    u64 add_timer(Timer&);
    u64 add_timer(u64 duration, TimeUnit, void (callback)());
    ...

private:
    SinglyLinkedList<Timer> m_timers;
};

In cpp file:
TimerQueue& TimerQueue::the()
{
    if (!queue)
        queue = new TimerQueue;
    return s_the;
}
```

Then you can use the single global instance of `TimerQueue` like so:

```cpp
TimerQueue::the().add_timer(...);
```